nemo/collections/asr/models/asr_model.py

```diff
@@ -89,3 +91,502 @@ def forward_for_export(self, input, length=None):
+    (covers most ASR classes)
+    """
+
+    @property
+    def input_module(self):
+        return self.encoder
+
+    @property
+    def output_module(self):
+        return self.decoder
+
+    @property
+    def joint_module(self):
+        return self.joint
+
+    @property
+    def disabled_deployment_input_names(self):
+        encoder_names = self.input_module.disabled_deployment_input_names
+        decoder_names = self.output_module.disabled_deployment_input_names
+        joint_names = self.joint_module.disabled_deployment_input_names
+        return set.union(encoder_names, decoder_names, joint_names)
+
+    @property
+    def disabled_deployment_output_names(self):
+        encoder_names = self.input_module.disabled_deployment_output_names
+        decoder_names = self.output_module.disabled_deployment_output_names
+        joint_names = self.joint_module.disabled_deployment_output_names
+        return set.union(encoder_names, decoder_names, joint_names)
+
+    def forward_for_encoder_export(self, encoder_inputs, length):
+        encoder_output = self.input_module(encoder_inputs, length)
+        encoder_output, encoded_length = encoder_output
+
+        return encoder_output, encoded_length
+
+    def forward_for_decoder_joint_export(self, encoder_output, decoder_inputs, decoder_lengths, state_h, state_c):
+        decoder, joint = self.output_module, self.joint_module
+
+        if state_h is not None and state_c is not None:
+            states = (state_h, state_c)
+        else:
+            states = None
+
+        decoder_outputs = decoder(decoder_inputs, decoder_lengths, states)
+        decoder_output = decoder_outputs[0]
+        decoder_length = decoder_outputs[1]
+        state_h, state_c = decoder_outputs[2][0], decoder_outputs[2][1]
+
+        joint_output = joint(encoder_output, decoder_output)
+        return (joint_output, decoder_length, state_h, state_c)
+
+    def export(
+        self,
+        output: str,
+        input_example=None,
+        output_example=None,
+        verbose=False,
+        export_params=True,
+        do_constant_folding=True,
+        keep_initializers_as_inputs=False,
+        onnx_opset_version: int = 13,
+        try_script: bool = False,
+        set_eval: bool = True,
+        check_trace: bool = False,
+        use_dynamic_axes: bool = True,
+        dynamic_axes=None,
+        check_tolerance=0.01,
+    ):
+        my_args = locals()
+        del my_args['self']
+
+        qual_name = self.__module__ + '.' + self.__class__.__qualname__
+        output_descr = qual_name + ' exported to ONNX'
+
+        try:
+            # Disable typechecks
+            typecheck.set_typecheck_enabled(enabled=False)
+
+            # Set module to eval mode
+            self._set_eval(set_eval)
+
+            format = self.get_format(output)
+
+            # Assign special flag for RNNT export of encoder
+            self.input_module._rnnt_export = True
```

Comment:
@borisfom here is where we set it explicitly to the modules. Needs to be before `prepare_for_export` since samples are already extracted by the time `prepare_for_extract` is called.